### DISCUSSION TITLE
Drone steadily losing and gaining height  after some time.

### DISCUSSION THREAD

**Discussion by abhilash1998** [iNavFlight/inav#7474] [Q&A]
We built a small quadcopter using KAKUTE F7  and Matek 3901-L0X Optical Flow & Lidar Sensor without a compass for autonomous flight. When we put the drone in ALT HOLD it behaves weirdly and accelerates in a random direction and hits the net. When put in loiter, the drone drifts a little in the horizontal plane and after some time it starts steadily losing the height and after some time it starts steadily gaining, but it is not an oscillation to consider it as a control problem. We found out that the altitude value from the matek sensor is pretty garbage. Hence we used TF mini as it is very accurate in altitude measurement, but it also has the same behavior.  We also tried tuning the PID parameters but it had no effect. After logging the data we observed the accelerometer values were pretty constant, which shows there is no effect of vibration The values of matek range-finder really fluctuated but for TF mini it was pretty constant. We tried tuning various parameters. The estimated value of height was varying. We reduced the weightage of the barometer and even nullified the values of the barometer. Another thing we tried was increasing the deadband but also it didn't work.
Can you please help me out solving this problem. Thank You
[LOG00006.TXT](https://github.com/iNavFlight/inav/files/7117762/LOG00006.TXT)

**Comment by capblack** [comment]
I experienced a similar problem but with almost total loss of control

https://www.youtube.com/watch?v=iWu_CB-ViNo

**Comment by DzikuVx** [comment]
![image](https://user-images.githubusercontent.com/966811/134664869-f702569a-badd-48e4-a5e8-87450aa14d6a.png)

The log shows that altitude follows target altitude no problems. However, rangefinder looses ground at around 1,5m of altitude

**Comment by anishparkhe1682** [comment]
hey @abhilash1998 i am a college student in pune. i am too working with matek optical sensor but facing some issues, can you pls help me
can you pls give me your email so that i can contact you
